lib/block-supports/layout.php

```diff
@@ -865,6 +865,20 @@ function gutenberg_restore_group_inner_container( $block_content, $block ) {
 		return $block_content;
 	}
 
+	$layout_classes = array();
+	$processor = new WP_HTML_Tag_Processor( $block_content );
+
+	if ( $processor->next_tag( array( 'class_name' => 'wp-block-group' ) ) ) {
+		foreach ( $processor->class_list() as $class_name ) {
+			if (str_contains( $class_name, 'layout' ) ) {
```

Comment:
It's probably unlikely that folks might use custom classnames that contain `layout`, but to be a bit more specific about it, would it be worth using `str_starts_with( 'is-layout-' )`?

Reply:
Oh wait, I forgot about the compound classnames like `wp-block-group-is-layout-flow` 🤔

Reply:
I think we'll have to take the risk with custom classnames 😅 because the crucial one for block gap to work is `wp-container-core-group-layout-x`. It was fortunate that #55416 had already made that classname layout-specific!